elements/rh-disclosure/rh-disclosure.ts

```diff
@@ -44,16 +56,34 @@ export class RhDisclosure extends LitElement {
 
   #onToggle(): void {
     this.open = this.detailsEl.open;
+    const event = new DisclosureToggleEvent();
+    this.dispatchEvent(event);
   }
 
   #onKeydown(event: KeyboardEvent): void {
+    const preventEscElements = `
```

Comment:
i think we'll need to use the event composed path for this

consider

```html
<rh-disclosure>
  <super-cheeky-element>
    #shadow-root
      <sc-input>
        #shadow-root
          <input>
```


so instead of `document.activeElement` (which might be a custom element with an arbitrary tag name, and we might not even be in the document while checking, we might be in another shadow root), so something like:

```ts
if (!event.composedPath().findLast(isLegitEscElement)) {
  // close...
}
```